File: Document/beginners/sample_mruby/sample02/app/walker.rb
class Walker

  PWM_MAX = 100
  PWM = PWM_MAX / 6
  DURATION = 2000

  # コンストラクタ
  #
  # 引数でポートを渡せる、省略時はデフォルトのポートを使う
  def initialize(left_motor_port=:port_c,
                 right_motor_port=:port_b,
                 direction)
    @left_wheel = EV3RT_TECS::Motor.new(left_motor_port)
    @right_wheel = EV3RT_TECS::Motor.new(right_motor_port)
    @forwarding = direction
  end

  def init
    Util::Display.init("sample02 " + self.class.to_s)
  end

  def terminate
    Util::Display.message("Stopped", 2)
    @right_wheel.stop
    @left_wheel.stop
  end
  
  def run
    if @forwarding
      Util::Display.message("Forwarding...", 2)
      @left_wheel.power = PWM
      @right_wheel.power = PWM
      @forwarding = false
    else
      Util::Display.message("Backwarding...", 2)
      @left_wheel.power = -PWM
      @right_wheel.power = -PWM
      @forwarding = true
    end
  end

end

File: Document/beginners/sample_java/sample02/src/jp/etrobo/ev3/sample02/Walker.java
/*
 *  Walker.java (for leJOS EV3)
 *  Created on: 2016/06/11
 *  Copyright (c) 2016 Embedded Technology Software Design Robot Contest
 */
package jp.etrobo.ev3.sample02;

import lejos.hardware.lcd.LCD;
import lejos.hardware.port.MotorPort;
import lejos.hardware.port.TachoMotorPort;

public class Walker {
    private TachoMotorPort leftWheel;
    private TachoMotorPort rightWheel;

	private final int pwm = TachoMotorPort.MAX_POWER / 6;
	private boolean isForwarding = true;

    Walker(){
    	leftWheel = MotorPort.C.open(TachoMotorPort.class);
    	rightWheel = MotorPort.B.open(TachoMotorPort.class);
    }

    public void init(){
    	LCD.drawString("Walker", 0, 2);
    }
    public void run(){
        if(isForwarding == true){
        	LCD.drawString("Forwarding...", 0, 4);
        	leftWheel.controlMotor(pwm, TachoMotorPort.PWM_FLOAT);
        	rightWheel.controlMotor(pwm, TachoMotorPort.PWM_FLOAT);
        	isForwarding = false;
        }else{
        	LCD.drawString("Backwarding...", 0, 4);
        	leftWheel.controlMotor(-pwm, TachoMotorPort.PWM_FLOAT);
        	rightWheel.controlMotor(-pwm, TachoMotorPort.PWM_FLOAT);
        	isForwarding = true;
        }
    }
    public void terminate(){
        LCD.drawString("Stopped.", 0, 4);
        leftWheel.setPWMMode(TachoMotorPort.STOP);
        rightWheel.setPWMMode(TachoMotorPort.STOP);

    }
}
